build_tools/install_rocm_from_artifacts.py

```diff
@@ -77,81 +81,44 @@ def _create_output_directory(args):
+
+    headers = {"Accept": "application/octet-stream"}
+
+    request = Request(s3_release_url, headers=headers)
+    with urlopen(request) as response, open(destination, "wb") as file:
+        if response.status == 404:
             log(
-                f"Error when retrieving GitHub release assets for release tag '{release_tag}'. This is most likely a rate limiting issue, so please try again"
+                f"S3 release assets for release bucket '{release_bucket}' and release version '{release_version}' not found."
             )
             return
         elif response.status != 200:
             log(
-                f"Error when retrieving GitHub release assets for release tag '{release_tag}' with status code {response.status}. Exiting..."
+                f"Error when retrieving S3 release assets for release bucket '{release_bucket}' and release version '{release_version}' with status code {response.status}. Exiting..."
             )
             return
 
-        release_data = json.loads(response.read().decode("utf-8"))
-
-    # We retrieve the most recent release asset that matches the amdgpu_family
-    # In the cases of "nightly-tarball" or "dev-tarball", this will retrieve the the specified version or latest
-    asset_data = sorted(
-        release_data["assets"], key=lambda item: item["updated_at"], reverse=True
-    )
-
-    # For nightly-tarball
-    if release_tag == "nightly-tarball" and release_version != "latest":
-        for asset in asset_data:
-            if amdgpu_family in asset["name"] and release_version in asset["name"]:
-                return asset
-    # For dev-tarball
-    elif release_tag == "dev-tarball":
-        for asset in asset_data:
-            if amdgpu_family in asset["name"] and release_version in asset["name"]:
-                return asset
-    # Otherwise, return the latest and amdgpu-matched asset available from the tag nightly-tarball
-    elif release_tag == "nightly-tarball" and release_version == "latest":
-        for asset in asset_data:
-            if amdgpu_family in asset["name"]:
-                return asset
-
-    return None
-
-
-def _download_github_release_asset(asset_data, output_dir):
-    """
-    With the GitHub asset data, this function downloads the asset to the output_dir
-    """
-    asset_name = asset_data["name"]
-    asset_url = asset_data["url"]
-    destination = output_dir / asset_name
-    headers = {"Accept": "application/octet-stream"}
-    # Making the API call to retrieve the asset
-    request = Request(asset_url, headers=headers)
-
-    with urlopen(request) as response_obj, open(destination, "wb") as file:
         # Downloading the asset to destination
         log(f"Downloading tar file to {str(destination)}")
-        shutil.copyfileobj(response_obj, file)
+        shutil.copyfileobj(response, file)
 
     # After downloading the asset, untar-ing the file
     _untar_files(output_dir, destination)
 
+    return None
+
```

Comment:
Is this `return None` needed? It looks like the other code paths just `return`, and this is at the end of the function anyways.